aroquev/src/main/webapp/script.js

```diff
@@ -192,29 +192,134 @@ function getRandomCar() {
-    // Get the cities container
-    const citiesContainer = document.getElementById('js-comments-container');
-    citiesContainer.innerHTML = '';
+function getServletComments() {
+  let maxComments = verifyNumberCommentsInput();
+
+  fetch('/data?maxComments=' + maxComments).then(response => response.json()).then((comments) => {
+    // Get the comments container
+    const commentsContainer = document.getElementById('js-comments-container');
+    commentsContainer.innerHTML = '';
     
     // Check if the array of comments is empty
     if (!Array.isArray(comments) || !comments.length) {
       const pElement = document.createElement('p');
       pElement.innerText = "Looks like there are no comments yet. Be the first one to comment!"
-      citiesContainer.appendChild(pElement);
+      commentsContainer.appendChild(pElement);
     } else {
       for (const comment of comments) {
-        // Add each comment as a <li> to the container
-        citiesContainer.appendChild(createListElement(comment));
+        commentsContainer.appendChild(createCommentElement(comment));
       }
     }
   });
 }
 
+/**
+ * Verify the input number from selection of number of comments to display
+ * Minimum number is 1 comment, and maximum is 10 comments
+ */
+function verifyNumberCommentsInput() {
+  // Default value
+  let maxComments = 5;
+
+  let maxCommentsInput = document.getElementById("maxComments").value;
+  const commentsErrorContainer = document.getElementById('js-comments-input-error-container');
+  if (maxCommentsInput.length != 0 && maxCommentsInput < 11 && maxCommentsInput > 0) {
+    maxComments = maxCommentsInput;
+    commentsErrorContainer.innerText = "";
+  }
+  else {
+    commentsErrorContainer.innerText = "Invalid input for number of comments!";
+  }
+
+  return maxComments;
+}
+
+/**
+ * Takes the comment object and builds the element in the DOM
+ * @param {*} comment the comment object to be displayed
+ */
+function createCommentElement(comment) {
+  const commentView = document.createElement('div');
+
+  const username = document.createElement('div');
+  username.innerText = `Posted by: ${comment.username}`;
+  username.classList.add('comment-username');
+
+  const timestamp = document.createElement('div');
+  const date = new Date(comment.timestamp);
+  const options = {
+    year: 'numeric',
+    month: 'short',
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric'
+  };
+  timestamp.innerText = `On: ${new Intl.DateTimeFormat('en', options).format(date)}`;
+  timestamp.classList.add('comment-timestamp');
+
+  const popularity = document.createElement('div');
+  popularity.innerText = `Popularity: ${comment.upvotes - comment.downvotes}`;
+
+  const body = document.createElement('div');
+  body.innerText = comment.body;
+  body.classList.add('comment-body');
+
+  const upvote = document.createElement('button');
+  upvote.innerText = "Upvote";
+  upvote.addEventListener('click', () => {
```

Comment:
Like I mentioned in the sync, the event listener could be removed after one click as a very simple way to deter spamming votes. Wouldn't stop users from just refreshing or writing a loop in the console, but it's better than nothing.

Reply:
Mmm I see. The thing is that because when the user hits the upvote comment, the getComments() method gets called and resets the whole section, so if I disable the clicking it gets enabled instantly because it's like the user never clicked anything. I did this to keep the DS as the source of data at all times, but now that's not helping in this situation. I was planning on kinda fixing this with the login auth, that would allow the user to either upvote or downvote the comment once.